### ISSUE TITLE
[Info] Migrate code from old AppBoy SDK to the new braze-swift-sdk

### ISSUE THREAD

**Issue by malcommac** [braze-inc/braze-swift-sdk#8]
I'm looking at the doc of the new Braze SDK, but I cannot find anything about migration from the [old AppBoy SDK](https://github.com/Appboy/appboy-ios-sdk).

Everything it's pretty straightforward, but I have the following doubts regarding Push Notification section:

- Is there an equivalent of the old `braze?.pushAuthorization(fromUserNotificationCenter:)` to inform about the status? Or it's unnecessary?
- Is there a new equivalent for `braze?.register(_:didReceiveRemoteNotification:fetchCompletionHandler:)` to inform Braze about upcoming notification? Is the `notifications.handleUserNotification()` and/or `handleBackgroundNotification()` methods? 

Finally, what's happened to the following methods of `ABKPushUtils`?  
They are very useful for knowing details about the push (we must support both native and braze push):

- `ABKPushUtils.isAppboyRemoteNotification`
- `ABKPushUtils.isAppboySilentRemoteNotification(dictionary)`
- `ABKPushUtils.isAppboyInternalRemoteNotification(dictionary)`

Thanks

**Comment by lowip** [comment]
Hi @malcommac, thanks for opening this issue.

We haven't released our migration guide yet, but we're working on it. You can check out our [_Upcoming Feature Roadmap_](https://github.com/braze-inc/braze-swift-sdk#upcoming-feature-roadmap) to keep track of the features we're working on.

To answer your questions:

> Is there an equivalent of the old `braze?.pushAuthorization(fromUserNotificationCenter:)` to inform about the status? Or it's unnecessary?

`braze?.pushAuthorization(fromUserNotificationCenter:)` is not needed anymore. The Braze SDK automatically reads that information after the user has interacted with the permission prompt.

> Is there a new equivalent for `braze?.register(_:didReceiveRemoteNotification:fetchCompletionHandler:)` to inform Braze about upcoming notification? Is the `notifications.handleUserNotification()` and/or `handleBackgroundNotification()` methods?

```swift
// appboy-ios-sdk:
braze?.register(_:didReceiveRemoteNotification:fetchCompletionHandler:)

// braze-swift-sdk:
braze?.notifications.handleBackgroundNotification(userInfo:fetchCompletionHandler:)
```

See the `handleBackgroundNotification(userInfo:fetchCompletionHandler:)` [documentation page][1] for information about proper usage.

Our [_Standard Push Notifications_][2] tutorial goes over all required methods for the push integrations.

> Finally, what's happened to the following methods of `ABKPushUtils`?

You may have noticed that our `notifications` push handling methods returns a boolean value. This return value indicates whether Braze handled the push or not.

If you need to handle push from other sources, you can do so by modifying the minimal implementations provided in these methods documentation. For instance, to handle background / silent pushes from other sources:

```swift
// In UIApplicationDelegate implementation.

func application(
  _ application: UIApplication,
  didReceiveRemoteNotification userInfo: [AnyHashable: Any],
  fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void
) {
  // Forward background notification to Braze.
  if braze.notifications.handleBackgroundNotification(
    userInfo: userInfo,
    fetchCompletionHandler: completionHandler
  ) {
   // Braze handled the notification, nothing more to do.
   return
 }

 // Braze did not handle this remote background notification.
 // → You can handle the notification yourself here.

 // Manually call the completion handler to let the system know
 // that the background notification is processed.
 completionHandler(.noData)
}
```

Is there any scenario where this approach would not be suitable for your application? We want to make sure how `isAppboyRemoteNotification`, `isAppboySilentRemoteNotification()` and `isAppboyInternalRemoteNotification()` are used and if they are even needed before offering them in the Swift SDK.

[1]: https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/notifications-swift.class/handlebackgroundnotification(userinfo:fetchcompletionhandler:)
[2]: https://braze-inc.github.io/braze-swift-sdk/tutorials/braze/b1-standard-push-notifications

**Comment by zizlak** [comment]
Hi @lowip , I 'm wondering if an installation guide is also available for cocoa pods? I could only find one for [swift packages](https://braze-inc.github.io/braze-swift-sdk/tutorials/braze/a1-installation), I could found this [podspec](https://github.com/CocoaPods/Specs/blob/master/Specs/4/0/b/BrazeKit/5.2.0/BrazeKit.podspec.json) but no instalation guide for cocoa pods.

**Comment by lowip** [comment]
Hi @zizlak, we do not have an installation guide for CocoaPods yet. Feel free to open an issue requesting one for better tracking.

You can use our [Examples](https://github.com/braze-inc/braze-swift-sdk/tree/main/Examples) project [Podfile](https://github.com/braze-inc/braze-swift-sdk/blob/main/Examples/Podfile) as a reference for your integration.

**Comment by malcommac** [comment]
Thank you, @lowip, I've correctly modified our handling of push, and everything seems okay.

However, I have a few other questions; I cannot find the `getDeviceId()` property in the new SDK. When we talked with the tech department, they said Braze SDK does not model the anonymous user.  
So, when our program moves to logged -> anonymous user, we add the current device id to the user's blocked devices list in order to prevent that device from receiving notifications from previously logged users.

That's the code:

```swift
    /// Add the device id to the current user's blacklisted devices.
    /// This prevent sending push notification to devices where the user is logged out.
    /// This value will be used in conjuction with Liquid template scripting to exclude previously logged devices.
    ///
    /// DISCUSSION:
    /// We can't move to modeled user to anonymous one because Braze does not have this concept.
    /// The only thing we can do is to make this device unreachable by the notifications send to this user.
    /// User on this device still be identified as the same old user by Braze but it will not receive notification
    /// as any other anonymous user.
    /// Frankly I don't know the real rational behind this but okay... that's how this stuff works.
    private func addDeviceToCurrentUserBlockedDevicesList() {
        guard let deviceIDFV = braze?.getDeviceId() else { return }

        braze?.user.addToCustomAttributeArray(withKey: LoggedOutDevicesAttributeKey, value: deviceIDFV)
    }
    
    /// This is the opposite operation of `addDeviceToCurrentUserBlockedDevicesList()`.
    /// When an user is logged in back with this device we want to be sure this device is not in its blacklisted devices list.
    /// This may happen if user make a login with a previously logged-out device.
    private func removeDeviceToCurrentUserBlockedDevicesList() {
        guard let deviceIDFV = braze?.getDeviceId() else { return }
        
        braze?.user.removeFromCustomAttributeArray(withKey: LoggedOutDevicesAttributeKey, value: deviceIDFV)
    }
```

Where is this property in the new SDK?

**Comment by malcommac** [comment]
Looking at the headers of the old SDK I found:

```objc
* Get the device ID - the IDFV - which will reset if all apps for a given vendor are removed from the device.
 - (NSString *)getDeviceId;
```

So I think it's pretty safe to use UIDevice.current.identifierForVendor!.uuidString

**Comment by lowip** [comment]
Hi @malcommac, sorry for the delayed response. We have released [5.3.1] which adds the missing [`Braze.deviceId(queue:completion:)`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/deviceid(queue:completion:)) method.

> So I think it's safe to use UIDevice.current.identifierForVendor!.uuidString

The device identifier is currently the `identifierForVendor` but this is subject to change in the future, I'd recommend updating your code to use `Braze.deviceId(queue:completion:)` instead.

I'm closing the issue, feel free to re-open it if needed.

**Comment by bfisherdk** [comment]
Hi, do you have any updated timeline on when the migration guide will be released?
